From bfaf7cb287a1c46733a079d20d14a017e81620e6 Mon Sep 17 00:00:00 2001
From: Rahul Sabnis <rahulsabnis@google.com>
Date: Wed, 6 Apr 2022 22:44:01 +0000
Subject: [PATCH] Removes app access to BluetoothAdapter#setDiscoverableTimeout
 by requiring BLUETOOTH_PRIVILEGED permission.

Bug: 206807679
Test: Manual
Change-Id: I73288f495d35280a5724d070248db54e2fe537fd
(cherry picked from commit cc22abc0643c66e5873cbd0d15cd128816a186cf)
Merged-In: I73288f495d35280a5724d070248db54e2fe537fd
---
 src/com/android/bluetooth/btservice/AdapterService.java | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/com/android/bluetooth/btservice/AdapterService.java b/src/com/android/bluetooth/btservice/AdapterService.java
index e6494ddab..c5c42201a 100644
--- a/src/com/android/bluetooth/btservice/AdapterService.java
+++ b/src/com/android/bluetooth/btservice/AdapterService.java
@@ -1460,6 +1460,8 @@ public class AdapterService extends Service {
                 return false;
             }
 
+            enforceBluetoothPrivilegedPermission(service);
+
             return service.mAdapterProperties.setDiscoverableTimeout(timeout);
         }
 
-- 
2.36.0.550.gb090851708-goog

